c10/util/Float8_e4m3fn.h

```diff
@@ -102,7 +103,9 @@ inline C10_HOST_DEVICE float fp8e4m3fn_to_fp32_value(uint8_t input) {
   _BitScanReverse(&nonsign_bsr, (unsigned long)nonsign);
   uint32_t renorm_shift = (uint32_t)nonsign_bsr ^ 31;
 #else
-  uint32_t renorm_shift = __builtin_clz(nonsign);
+  // Note: zero is not a supported input into `__builtin_clz`
```

Comment:
for posterity:
* failing job: https://github.com/pytorch/pytorch/actions/runs/5647463609/job/15297976230?pr=105807
* error message:

```
  test_quantization.py::TestFloat8DtypeCPU::test_creation_with_zeros_float8_e4m3fn_cpu_float8_e4m3fn <- test/quantization/core/experimental/test_float8.py /var/lib/jenkins/workspace/c10/util/Float8_e4m3fn.h:105:41: runtime error: passing zero to clz(), which is not a valid argument
      #0 0x7fa85916471d in void at::vec::AVX2::convert<c10::Float8_e4m3fn, float>(c10::Float8_e4m3fn const*, float*, long) (/opt/conda/envs/py_3.9/lib/python3.9/site-packages/torch/lib/libtorch_cpu.so+0x1ba0a71d)
      #1 0x7fa85905d660 in void c10::function_ref<void (char**, long const*, long, long)>::callback_fn<auto at::TensorIteratorBase::loop_2d_from_1d<at::native::AVX2::copy_kernel(at::TensorIterator&, bool)::$_3::operator()() const::'lambda5'()::operator()() const::'lambda'()::operator()() const::'lambda13'()::operator()() const::'lambda'(char**, long const*, long)>
```

I wasn't able to get a local repro, but we'll see if the below fix fixed CI